File: src/webserver/viewModels/books.go
package viewModels

import "webserver/model"

type BookVm struct {
	Title string
	Books []model.Book
}
func NewBooks(books []model.Book) BookVm {
	return BookVm{
		Title: "Awesome Books",
		Books: books,
	}
}

type Category struct {
	Name string
}

type Book struct {
	Id int `json: "id"`
	Title string `json: "title"`
	Img string `json: "img"`
	Categories[] Category `json: "categories"`
	Description string `json: "description"`
	Published string `json: "published"`
}

File: public/js/books.js
(function() {
    var data = {
        id: 3,
        title: 'Mastering Kubernetes',
        img: 'https://images-na.ssl-images-amazon.com/images/I/41deaLZesLL._SX404_BO1,204,203,200_.jpg',
        categories: [{
            name: 'Software Development'
        }],
        description: 'This practical guide demystifies Kubernetes and ensures that your clusters are always ' +
        'available, scalable, and up to date',
        published: 'May 25, 2017'
    };

    var bookBtn = document.getElementById('add-book');

    bookBtn.addEventListener('click', function() {
       fetch('/api/books', {
           method: 'post',
           body: JSON.stringify(data)
       });
    });
})();